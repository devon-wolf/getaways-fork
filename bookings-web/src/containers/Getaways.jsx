import React, { useState, useEffect } from 'react';
import { getPlaces } from '../services/placesApi';
import PlaceList from '../components/places/PlaceList';

const Getaways = () => {
  const [places, setPlaces] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    getPlaces()
      .then(setPlaces)
      .finally(() => setLoading(false));
  }, []);

  return (
    <>
    {loading && <p>Loading</p>}
    <PlaceList places={places} />
    </>
  );
};

export default Getaways;
