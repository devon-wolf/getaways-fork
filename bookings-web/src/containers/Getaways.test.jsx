import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import Getaways from './Getaways';

describe('Getaways page', () => {
	it('Renders the getaways page', () => {
		render(<Getaways />);
		screen.getByText('Loading');

		return waitFor(() => {
			const placeList = screen.getByLabelText('place list');
			expect(placeList).not.toBeEmptyDOMElement();
			expect(placeList).toMatchSnapshot();
		});
	});
})