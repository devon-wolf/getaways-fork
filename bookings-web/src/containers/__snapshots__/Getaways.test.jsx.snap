// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Getaways page Renders the getaways page 1`] = `Promise {}`;
